opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/reminder/SyncAccountAlarmListener.java

```diff
@@ -0,0 +1,33 @@
+package de.geeksfactory.opacclient.reminder;
+
+import com.commonsware.cwac.wakeful.WakefulIntentService;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+
+public class SyncAccountAlarmListener implements WakefulIntentService.AlarmListener {
+    // We could use a longer interval like 3 hours, but on API < 19, only the AlarmManager
+    // .INTERVAL_... constants make the AlarmManager use inexact alarm delivery for better
+    // battery efficiency. The next larger constant available would be AlarmManager
+    // .INTERVAL_HALF_DAY.
```

Comment:
> If we have no internet connection (or no WiFi with the corresponding setting activated, or a library website is offline) for a longer time (say, a week), the service would run every (inexact) half day and schedule itself to run one hour later. Then, the "one hour later" runs would reschedule themselves to run one hour later and won't stop doing that when the next "every half day" run is reached. This would result in the service running ~14 times per hour at the end of the week (and probably not concentrated around one point in the hour due to the inexactness of the "every half day" alarms).

This should, of course, not happen, but there are things we can do to prevent it. We only need a way for the scheduled run to determine if there already is an hourly run scheduled and only reschedule itself on failure if there isn't, right?

>We could also use Android's JobScheduler instead of AlarmManager for scheduling service runs, so it won't even run when there's no Internet connection available. The problem is it is only available on API 21 and above. There's also GcmNetworkManager, but it only works on devices with Google Play Services.

Sadly, this is both not an option currently. :(